src/domain/app/AppConfig.ts

```diff
@@ -11,15 +12,38 @@ class AppConfig {
     return new URL(this.origin).hostname;
   }
 
-  static get ApiUrl() {
+  static get apiUrl() {
     return getEnvOrError(import.meta.env.VITE_API_URI, 'VITE_API_URI');
   }
 
   static get oidcAuthority() {
-    return getEnvOrError(
+    const origin = getEnvOrError(
       import.meta.env.VITE_OIDC_AUTHORITY,
       'VITE_OIDC_AUTHORITY'
     );
+    return new URL(origin).href;
+  }
+
+  static get oidcClientId() {
+    return getEnvOrError(
+      import.meta.env.VITE_OIDC_CLIENT_ID,
+      'VITE_OIDC_CLIENT_ID'
+    );
+  }
+
+  static get oidcScope() {
+    return getEnvOrError(import.meta.env.VITE_OIDC_SCOPE, 'VITE_OIDC_SCOPE,');
+  }
+
+  static get oidcKukkuuApiClientId() {
+    return getEnvOrError(
+      import.meta.env.VITE_OIDC_KUKKUU_API_CLIENT_ID,
+      'VITE_OIDC_KUKKUU_API_CLIENT_ID'
+    );
+  }
+
+  static get oidcServerType(): 'KEYCLOAK' | 'TUNNISTAMO' {
```

Comment:
Adding type checking:
```typescript
  static get oidcServerType(): OidcServerType {
    const oidcServerType = import.meta.env.VITE_OIDC_SERVER_TYPE ?? 'KEYCLOAK';
    if (!OIDC_SERVER_TYPES.includes(oidcServerType)) {
      throw new Error(`Invalid OIDC server type: ${oidcServerType}`);
    }
    return oidcServerType;
  }
```
with
```typescript
const OIDC_SERVER_TYPES = ['KEYCLOAK', 'TUNNISTAMO'] as const;
type OidcServerType = (typeof OIDC_SERVER_TYPES)[number];
```
somewhere.

Reply:
But it's a bit hard to determine if the types really work as the "yarn typecheck" doesn't say anything about the AppConfig.ts file. I changed the code something that definitely should've broken the types and still "yarn typecheck" passed fine.

Reply:
Well, as you are typing the return value to be either `KEYCLOAK` or `TUNNISTAMO` without checking that they actually are then the type system is fooled here into showing stuff that isn't really necessarily true. So either removing the return typing (worse IMO) or making the necessary checks in order to really to be able to type the return value as either `KEYCLOAK` or `TUNNISTAMO` would be good. And if "yarn typecheck" is not working for AppConfig.ts then it would be better to fix that so it does work.

Reply:
I think the version we came up together in a huddle is ok now.